core/src/main/java/jenkins/model/queue/AsynchronousExecution.java

```diff
@@ -106,13 +107,36 @@ public synchronized final Executor getExecutor() {
         return executor;
     }
 
-    @Restricted(NoExternalUse.class)
+    /**
+     * Use {@link #setExecutorWithoutCompleting(Executor)} and {@link #maybeComplete()}
+     * instead. For this method, locking behaviour is problematic.
+     */
+    @Restricted(DoNotUse.class) @Deprecated()
```

Comment:
If the method is no longer called within this repository, you may simply delete it. That is the benefit of `NoExternalUse`!